### PR TITLE
Patch libevent

### PR DESCRIPTION
**Pull request by lasradoVinod**
Using a custom libevent makes it difficult for other projects to use. This code change is a way to help continue use the standard libevent.

### CODE REVIEW

File: net_http/server/internal/evhttp_server.cc [hunk 1]
```diff
@@ -405,4 +405,8 @@ bool EvHTTPServer::EventLoopSchedule(std::function<void()> fn) {
   return result == 0;
 }
 
+void EvHTTPServer::ScheduleReply(std::function<void()> fn) {
+  server_options_->executor()->Schedule(fn);
```
Comment (wenbozhu) [inline net_http/server/internal/evhttp_server.cc:409]: why does this become safe now, i.e. scheduled outside the event loop?

Comment (lasradoVinod) [reply net_http/server/internal/evhttp_server.cc:409]: It is mentioned everywhere that libevent does not handle deffered replies well:
1. https://github.com/libevent/libevent/issues/944
2. https://github.com/azat/libevent/commit/5ff8eb26371c4dc56f384b2de35bea2d87814779. 

It is a simpler solution to reply inline.

Comment (wenbozhu) [reply net_http/server/internal/evhttp_server.cc:409]: Thanks! I don't think I know libevents any more :)

Comment (wenbozhu) [reply net_http/server/internal/evhttp_server.cc:409]: LGTM.

Comment (lasradoVinod) [reply net_http/server/internal/evhttp_server.cc:409]: Can you please approve it as well?

**Review by wenbozhu — APPROVED** [review]

**Pull request merged**